api/v1alpha4/clusterclass_webhook.go

```diff
@@ -19,16 +19,24 @@ package v1alpha4
 import (
 	"fmt"
 	"reflect"
+	"strings"
 
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 	"sigs.k8s.io/cluster-api/feature"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
 )
 
+const (
+	// TemplateSuffix is the object kind suffix used by infrastructure references associated
+	// with MachineSet or MachineDeployments.
```

Comment:
```suggestion
	// TemplateSuffix is the object kind suffix used by template types.
	TemplateSuffix = "Template"
```
nit (it's not only infra)

Reply:
Should we maybe move this constant to common_types.go?